tiledb/api/src/query/read/aggregate/mod.rs

```diff
@@ -0,0 +1,1054 @@
+                    ctx,
+                    core::ptr::addr_of_mut!(c_agg_operation)
+                        as *mut *const tiledb_channel_operation_t,
+                )
+            })?;
+        } else {
+            let c_field_name: *const i8 =
+                handle.field_name.as_ref().unwrap().as_c_str().as_ptr();
+            match handle.function {
+                AggregateFunction::Count => unreachable!(
+                    "AggregateFunction::Count handled in above case, found {:?}",
+                    handle.function
+                ),
+                AggregateFunction::NullCount(_) => {
+                    context.capi_call(|ctx| unsafe {
+                        ffi::tiledb_channel_operator_null_count_get(
+                            ctx,
+                            &mut c_agg_operator,
+                        )
+                    })?;
+                }
+                AggregateFunction::Sum(_) => {
+                    context.capi_call(|ctx| unsafe {
+                        ffi::tiledb_channel_operator_sum_get(
+                            ctx,
+                            &mut c_agg_operator,
+                        )
+                    })?;
+                }
+                AggregateFunction::Max(_) => {
+                    context.capi_call(|ctx| unsafe {
+                        ffi::tiledb_channel_operator_max_get(
+                            ctx,
+                            &mut c_agg_operator,
+                        )
+                    })?;
+                }
+                AggregateFunction::Min(_) => {
+                    context.capi_call(|ctx| unsafe {
+                        ffi::tiledb_channel_operator_min_get(
+                            ctx,
+                            &mut c_agg_operator,
+                        )
+                    })?;
+                }
+                AggregateFunction::Mean(_) => {
+                    context.capi_call(|ctx| unsafe {
+                        ffi::tiledb_channel_operator_mean_get(
+                            ctx,
+                            &mut c_agg_operator,
+                        )
+                    })?;
+                }
+            };
+            context.capi_call(|ctx| unsafe {
+                ffi::tiledb_create_unary_aggregate(
+                    ctx,
+                    c_query,
+                    c_agg_operator,
+                    c_field_name,
+                    &mut c_agg_operation,
+                )
+            })?;
+        }
+
+        context.capi_call(|ctx| unsafe {
+            ffi::tiledb_channel_apply_aggregate(
+                ctx,
+                c_channel,
+                c_agg_name,
+                c_agg_operation,
+            )
+        })?;
+
+        Ok(AggregateBuilder::<T, Self> {
+            base: self,
+            handle,
+            field_type: PhantomData,
+        })
+    }
+
+    /// Adds a request for the number of results which satisfy the query predicates.
```

Comment:
Does core return a count of matching results when given a query condition? I only vaguely paid attention when this was being implemented and I don't remember if that was supported or if it was just about "number of cells in the subarray ranges provided".

Reply:
Confirmed via example that changing the subarray changes aggregate results